libminifi/src/RemoteProcessorGroupPort.cpp

```diff
@@ -84,19 +107,41 @@ void RemoteProcessorGroupPort::initialize() {
   std::set<core::Relationship> relationships;
   relationships.insert(relation);
   setSupportedRelationships(relationships);
+  curl_global_init(CURL_GLOBAL_DEFAULT);
+  site2site_peer_mutex_.lock();
```

Comment:
I agree this seems to be a good use case for lock_guard as the lock is obtained at the start of initialize() and released just before returning.